packages/aws-cdk-lib/aws-ec2/test/vpc.test.ts

```diff
@@ -719,13 +719,38 @@ describe('vpc', () => {
 
     test('with availabilityZones set to zones different from stack', () => {
       const stack = getTestStack();
+      // we need to create a context with availability zones, otherwise we're checking against dummy values
+      stack.node.setContext(
+        `availability-zones:account=${stack.account}:region=${stack.region}`,
+        ['us-east-1a', 'us-east1b', 'us-east-1c'],
+      );
```

Comment:
The reason for this change in an existing test is because we need to check the provided availability zones against well-defined availability zones, i.e., from a context. If availability zones aren't provided in a context, then we're checking against tokens or 'dummy' values which is unnecessary. With the changes I made, this test no longer throws because we've eliminated any validation when 'dummy' values are returned. Here, I've added AZs in a context so that we're still checking provided AZs against stack AZs.